### DISCUSSION TITLE
How can I get each turbine wake loss and NET AEP in a farm ?

### DISCUSSION THREAD

**Discussion by hkashiw** [NatLabRockies/floris#708] [Q&A] [answered]
How can I obtain the wake loss for each individual turbine in a wind farm?
Also, could you tell us how you calculate AEP including wake loss?

Thank you for your help,

**Comment by Bartdoekemeijer** [comment, marked as answer]
Hi @hkashiw,

Have you tried studying the examples that come with FLORIS: https://github.com/NREL/floris/tree/main/examples? Examples 04 to 06 explain how to calculate the turbine power productions for various wind directions and wind speeds, including their wake losses. Examples 07 and 08 explain how to calculate the AEP of a wind farm for a prespecified wind rose, including wake losses.

Best,
Bart

**Reply by hkashiw** [reply]
Hi Bartdoekemeijer,

Thank you for contacting us.
I have checked the example.
I ran 08 and it outputs the AEP for the entire farm, but not the AEP or wake loss for each wind turbine.
Here is an image of the output I want.

Turbine		Gross AEP(no wake)	wake loss	　　NET AEP		Wind Speed
T01		**			**		**		**
T02		**			**		**		**
T03		**			**		**		**

**Reply by Nellytado** [reply]
Hi, when calculating the AEP for a farm, if we have our own wind rose data with a list of wind speeds and wind directions how can we obtain the values for freq_val? Is there a code we can use to generate a separate file csv if we have data under the headings: YYYYMMDD,HHMM,M(m/s),D(deg),T(C),De(k/m3),PRE(hPa),RiNumber,RH(%),RMOL(1/m)? Thank you any help is much appreciated!
